dascore/io/rsf/core.py

```diff
@@ -0,0 +1,100 @@
+"""IO module for reading and writing RSF file format support of MADAGASCAR."""
+# from __future__ import annotations
+
+# import segyio
+import datetime as dt
+from pathlib import Path
+
+import numpy as np
+
+import dascore as dc
+from dascore.io.core import FiberIO
+from dascore.utils.time import to_float
+
+RSFKEYS_WRITE = ("in", "esize", "data_format")
+
+
+class RSFV1(FiberIO):
+    """An IO class supporting the RSF format of MADAGASCAR."""
+
+    name = "rsf"
+    preferred_extensions = "rsf"
+    # also specify a version so when version 2 is released you can
+    # just make another class in the same module named rsfV2.
+    version = "1"
+
+    ########################
+    # Add a way to read in rsf files.
+    # May need a way to get absolute time stamp,
+    #   like hide it in header file when writing out an RSF file.
+    ########################
+
+    def write(self, spool, path, data_path=None, **kwargs):
+        """
+        Write a patch to RSF format.
+
+        if no data_path is NOT specified, the header and binary will be
+        packed together
+        data_path needs to be bindata_file.rsf or /location/of/bindata_file.rsf
+        (NO '@')
+
+        path needs to be hdr_file.rsf or /location/of/hdr_file.rsf
+
+        spool needs to have a single patch in it
```

Comment:
So this will generate a nice API doc page, we need a parameter section:

```python

Parameters
----------------
spool
    The input spool to convert to rsf, must have exactly one patch. 
path
    Path to create the rsf file
data_path
    If data and rsf header information are to be separate, the location of the data file. Needs to be 
    bindata_file.rsf or /location/of/bindata_file.rsf (no '@')

Notes
--------
- Patch datatype is converted to float32 for compatibility with Madagascar (may be able to keep dytpe in the future)
```